docs/src/main/paradox/scala/http/routing-dsl/directives/index.md

```diff
@@ -146,31 +210,56 @@ transformations, both (or either) on the request and on the response side.
 
 ## Composing Directives
 
-@@@ note
+@@@ note { .group-scala }
 Gotcha: forgetting the `~` (tilde) character in between directives can result in perfectly valid
 Scala code that compiles but does not work as expected. What would be intended as a single expression would actually be multiple expressions, and only the final one would be used as the result of the parent directive. Alternatively, you might choose to use the `concat` combinator. `concat(a, b, c)` is the same as `a ~ b ~ c`.
 @@@
 
 As you have seen from the examples presented so far the "normal" way of composing directives is nesting.
 Let's take a look at this concrete example:
 
-@@snip [DirectiveExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/DirectiveExamplesSpec.scala) { #example-1 }
+Scala
+:  @@snip [DirectiveExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/DirectiveExamplesSpec.scala) { #example-1 }
+
+Java
+:  @@snip [DirectiveExamplesTest.java]($test$/java/docs/http/javadsl/server/DirectiveExamplesTest.java) { #example1 }
+
+Here the `get` and `put` directives are chained together @scala[with the `~` operator]@java[using the `orElse` method] to form a higher-level route that
```

Comment:
eventually start recommending `concat()` for scala I guess. But don't mind right now, let's just get them all consolidated first